generator/src/main/java/com/graphicsfuzz/generator/tool/ShaderGenerator.java

```diff
@@ -0,0 +1,132 @@
+import com.graphicsfuzz.common.ast.decl.PrecisionDeclaration;
+import com.graphicsfuzz.common.ast.decl.VariableDeclInfo;
+import com.graphicsfuzz.common.ast.decl.VariablesDeclaration;
+import com.graphicsfuzz.common.ast.expr.BinOp;
+import com.graphicsfuzz.common.ast.expr.BinaryExpr;
+import com.graphicsfuzz.common.ast.expr.Expr;
+import com.graphicsfuzz.common.ast.expr.TypeConstructorExpr;
+import com.graphicsfuzz.common.ast.expr.VariableIdentifierExpr;
+import com.graphicsfuzz.common.ast.stmt.BlockStmt;
+import com.graphicsfuzz.common.ast.stmt.DeclarationStmt;
+import com.graphicsfuzz.common.ast.stmt.ExprStmt;
+import com.graphicsfuzz.common.ast.type.BasicType;
+import com.graphicsfuzz.common.ast.type.VoidType;
+import com.graphicsfuzz.common.glslversion.ShadingLanguageVersion;
+import com.graphicsfuzz.common.tool.PrettyPrinterVisitor;
+import com.graphicsfuzz.common.util.GlslParserException;
+import com.graphicsfuzz.common.util.IRandom;
+import com.graphicsfuzz.common.util.ParseTimeoutException;
+import com.graphicsfuzz.common.util.PipelineInfo;
+import com.graphicsfuzz.common.util.RandomWrapper;
+import com.graphicsfuzz.generator.mutateapi.ExpressionGenerator;
+import com.graphicsfuzz.generator.mutateapi.FactManager;
+import com.graphicsfuzz.generator.mutateapi.NumericValue;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Optional;
+import net.sourceforge.argparse4j.inf.ArgumentParserException;
+
+public class ShaderGenerator {
+
+  public static void mainHelper(String[] args) throws ArgumentParserException, IOException,
+      ParseTimeoutException, InterruptedException, GlslParserException {
+
+    final TranslationUnit tu = new TranslationUnit(Optional.of(ShadingLanguageVersion.GLSL_410),
+        Arrays.asList(
+            new PrecisionDeclaration("precision mediump float;"),
+            new FunctionDefinition(
+                new FunctionPrototype("main", VoidType.VOID, Collections.emptyList()),
+                new BlockStmt(new ArrayList<>(), false))));
+
+    final PipelineInfo pipelineInfo = new PipelineInfo();
+    final IRandom generator = new RandomWrapper(0);
+    final FactManager globalFactManager = new FactManager(null);
+
+    final ExprStmt colorAssignment = new ExprStmt(null);
+
+    ExpressionGenerator expressionGenerator = new
+        ExpressionGenerator(tu, pipelineInfo, generator, globalFactManager);
+    tu.getMainFunction().getBody().addStmt(new DeclarationStmt(
+        new VariablesDeclaration(BasicType.VEC4,
+            new VariableDeclInfo(
+                "_GLF_color",
+                null,
+                null
+            )
+        )
+    ));
+    tu.getMainFunction().getBody().addStmt(colorAssignment);
+    final FactManager mainFactManager = globalFactManager.newScope();
+    final Expr rvalue = expressionGenerator.generateExpr(
+        mainFactManager,
+        tu.getMainFunction(),
+        colorAssignment,
```

Comment:
Please replace all four of 0.0, 1.0, 0.3, 1.0 with things that can come from the command line.